mmv1/products/apigee/ApiProduct.yaml

```diff
@@ -0,0 +1,423 @@
+
+                Note: Currently, you can specify only a single GraphQLOperation. Specifying more than one will cause the operation to fail.
+              item_type:
+                type: String
+            - name: "quota"
+              type: NestedObject
+              description: |
+                Quota parameters to be enforced for the resources, methods, and API source combination. If none are specified, quota enforcement will not be done.
+              properties:
+                - name: "limit"
+                  type: String
+                  description: |
+                    Required. Upper limit allowed for the time interval and time unit specified. Requests exceeding this limit will be rejected.
+                - name: "interval"
+                  type: String
+                  description: |
+                    Required. Time interval over which the number of request messages is calculated.
+                - name: "timeUnit"
+                  type: String
+                  description: |
+                    Time unit defined for the interval. Valid values include second, minute, hour, day, month or year. If limit and interval are valid, the default value is hour; otherwise, the default is null.
+            - name: "attributes"
+              type: Array
+              is_set: true
+              description: |
+                Custom attributes associated with the operation.
+              item_type:
+                type: NestedObject
+                properties:
+                  - name: "name"
+                    type: String
+                    description: |
+                      Key of the attribute.
+                  - name: "value"
+                    type: String
+                    description: |
+                      Value of the attribute.
+            - name: "service"
+              type: String
+              description: |
+                Required. gRPC Service name associated to be associated with the API proxy, on which quota rules can be applied upon.
+
+  - name: "quotaCounterScope"
+    type: Enum
+    description: |
+      Scope of the quota decides how the quota counter gets applied and evaluate for quota violation. If the Scope is set as PROXY, then all the operations defined for the APIproduct that are associated with the same proxy will share the same quota counter set at the APIproduct level, making it a global counter at a proxy level. If the Scope is set as OPERATION, then each operations get the counter set at the API product dedicated, making it a local counter. Note that, the QuotaCounterScope applies only when an operation does not have dedicated quota set for itself.
+    enum_values:
+      - "QUOTA_COUNTER_SCOPE_UNSPECIFIED"
+      - "PROXY"
+      - "OPERATION"
+
+  - name: "space"
```

Comment:
If `space` isn't supported by the API yet it should be removed from the PR and added later when API support is added (`immutable` doesn't make a field unsettable. it means "delete the resource and recreate when this field is changed")

Reply:
It works when you provide the "space" value calling manually, I just couldn't write a test because we don't have it as a Terraform resource yet (the Space resource).

Regarding the immutable, the space is supposed to be immutable, it can't be changed once you create the API Product resource.